tls/s2n_ktls_io.c

```diff
@@ -172,3 +173,40 @@ S2N_RESULT s2n_ktls_get_control_data(struct msghdr *msg, int cmsg_type, uint8_t
 
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_ktls_sendmsg(struct s2n_connection *conn, uint8_t record_type, const struct iovec *msg_iov,
+        size_t msg_iovlen, s2n_blocked_status *blocked, size_t *bytes_written)
+{
+    RESULT_ENSURE_REF(bytes_written);
+    RESULT_ENSURE_REF(msg_iov);
+    RESULT_ENSURE_REF(blocked);
+    RESULT_ENSURE_REF(conn);
+
+    *blocked = S2N_BLOCKED_ON_WRITE;
+
+    RESULT_ENSURE(msg_iov->iov_base, S2N_ERR_INVALID_ARGUMENT);
```

Comment:
Why is this check necessary?

Reply:
We are creating a msghdr using iovec and there is an assumption that iovec is not malformed. This check confirms that assumption. If iovec is NULL then msghdr is also malformed.

Reply:
Would the iovec be considered malformed if iov_base is NULL and iov_len is 0?

Reply:
ill check and ban both `NULL and 0`.. that seems safer for now and we can support sending 0 later if we need